manifests/repo/debian.pp

```diff
@@ -37,20 +33,24 @@
 ) {
   assert_private()
 
+  if $facts['os']['name'] != 'Debian' {
+    fail("class php::repo::debian does not work on OS ${facts['os']['name']}")
+  }
   include 'apt'
 
-  apt::source { "source_php_${release}":
-    location => $location,
-    release  => $release,
-    repos    => $repos,
-    include  => {
-      'src' => $include_src,
-      'deb' => true,
-    },
-    key      => $key,
+  if ($dotdeb and $facts['os']['release']['major'] in ['6', '7', '8']) {
```

Comment:
i personally would go for class parameter instead of magic numbers.

`data/default.yaml`:
```
...
php::repo::debian::dotdeb_available_for_releases: [6, 7, 8]
php::repo::debian::sury_available_for_releases: [9, 10, 11]
...
```

```
class php::repo::debian (
  Array[Integer] $dotdeb_available_for_releases,
  Array[Integer] $sury_available_for_releases,
...
) {
  ...
  if ($dotdeb and Integer($facts['os']['release']['major']) in $dotdeb_available_for_releases) {
  ...
  if ($sury and Integer($facts['os']['release']['major']) in $sury_available_for_releases) {
  ...
}
```

for me the functionality is clearer documented and the end user can still change the behavior by overriding the variables. for example if debian 12 is out and this module doesn't get updated fast enough, the users can define their own variable and add `12` to it without waiting for the module being fixed.

Reply:
Hello! Looks interesting. Will we also need to update the documentation in that case? Thanks!

Reply:
Hello! Today I was re-installing the module and got the issue back. Will we merge this or need more rework?